services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/permissions/HankeKayttaja.kt

```diff
@@ -146,6 +146,19 @@ class HankekayttajaEntity(
         yhteyshenkilot.map { it.hankeYhteystieto.contactType }
 
     fun fullName() = listOf(etunimi, sukunimi).filter { it.isNotBlank() }.joinToString(" ")
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as HankekayttajaEntity
+
+        return sahkoposti == other.sahkoposti
+    }
+
+    override fun hashCode(): Int {
+        return sahkoposti.hashCode()
```

Comment:
This will fail if at some point there's a set of hankekayttaja from different hanke. The default hashCode (instance-based) is safer than this.

Or you could map the entities to domain objects before creating a set.

Reply:
I changed to use default hashCode/equals.

Reply:
This still makes me a bit uneasy. I would avoid using sets here. But if you're convinced this works correctly, you can keep this like it is.